src/application/commands.rs

```diff
@@ -71,6 +73,26 @@ impl Application {
                 }
             }
             ui::draw(&mut self.terminal, &state)?;
+            // check for ctrl_c
+            if let Some(Event::Terminal(TermEvent::Key(KeyEvent {
```

Comment:
After calling `receive_event` you will consume the event even if this event is not a`Event::Terminal(TermEvent::Key(KeyEvent))`, so any other event is lost here. The `receive_event` method should be called only from one place.

Reply:
I think while sending we should ignore all other events.

I think it will be really weird trying to synchronise between other events and sending

Like if we receive a message while we're sending what should happen then

What about storing all not handled events and flush them after finishing the sending?

Reply:
> What about storing all not handled events and flush them after finishing the sending?

This is the current behavior without this PR. Terminal events will be processed at the end. While you are in the command loop sending the message, the TerminalCollector is sending into the event queue the terminal events.